### PR TITLE
Feature/smoke test forms 537

### PR DESCRIPTION
**Pull request by TheSpartan1980**
# Description
Created Scenarios for:

- Editing links
- Deleting a link between two pages
- Define a new condition when editing links
- Create a condition and add it to a link

## Type of change

Please delete options that are not relevant.

- [x] Automated test

# How Has This Been Tested?

The test was created using the red-green-refactor cycle, the test has also been run in Chrome and Chrome headless locally

# Checklist:

- [x] My changes do not introduce any new linting errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and versioning
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased onto master and there are no code conflicts
- [ ] I have checked deployments are working in all environments
- [ ] I have updated the architecture diagrams as per Contribute.md

### CODE REVIEW

File: designer/client/components/Visualisation/Lines.tsx [hunk 1]
```diff
@@ -16,51 +17,59 @@ type State = {
 };
 
 export class Lines extends React.Component<Props, State> {
+  static contextType = DataContext;
```
Comment (endamccormack) [inline designer/client/components/Visualisation/Lines.tsx:20]: never used?

File: designer/client/components/Visualisation/Lines.tsx [hunk 2]
```diff
@@ -16,51 +17,59 @@ type State = {
...
 
     return (
       <div>
         <svg height={layout.height} width={layout.width}>
           {layout.edges.map((edge) => {
-            const points = edge.points
-              .map((points) => `${points.x},${points.y}`)
-              .join(" ");
+            const { source, target, points, label } = edge;
+            const pointsString = points.map((p) => `${p.x},${p.y}`).join(" ");
```
Comment (endamccormack) [inline designer/client/components/Visualisation/Lines.tsx:45]: are you guaranteed points will never be null/undefined?

File: smoke-tests/designer/data/componentFieldData.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ module.exports = {
           You can see the results as soon as the MOT centre has recorded the test result.\n
           You’ll need the 11-digit number from the vehicle’s log book (V5C) to see the test location.`,
   },
-  yesNo: {
+  yesno: {
```
Comment (endamccormack) [inline smoke-tests/designer/data/componentFieldData.js:38]: this seems less consistent

**Review by endamccormack — CHANGES_REQUESTED** [review]

File: designer/client/components/Visualisation/Lines.tsx (hunk 1 above)
Comment (jenbutongit) [reply designer/client/components/Visualisation/Lines.tsx:20]: It is, you need to define contextType for class components so you can use this.context

File: designer/client/components/Visualisation/Lines.tsx (hunk 2 above)
Comment (jenbutongit) [reply designer/client/components/Visualisation/Lines.tsx:45]: No, but these are values provided by a library. This is unchanged from the code that was previously there, I've just cleaned it up a bit with a destructure

File: smoke-tests/designer/data/componentFieldData.js (hunk 3 above)
Comment (TheSpartan1980) [reply smoke-tests/designer/data/componentFieldData.js:38]: @endamccormack an unfortunate change I had to make as it appears in the create component section as `YesNo` rather then `Yes No` so the only option was to change the case or come up with a special function that only gets used for one component.

**Review by endamccormack — APPROVED** [review]

**Pull request merged**